macos/create_installer_macos.sh

```diff
@@ -0,0 +1,431 @@
+  else
+    echo "INFO: File '$1' with type '${type}' ignored."
+  fi
+}
+
+# Add files from a MacPorts package using package name and grp filter
+# $1 = MacPorts package name
+# $2 = regexp filter (grep)
+# Note:
+# This function strips the install path of the "port" command
+
+function add_files_using_macports_package {
+  echo "Copying files from MacPorts package $1 ..."
+  for file in $(port contents "$1" | grep "$2" | sort -u)
+  do
+    relpath="${file#${PORTDIR}}"
+    reldir="${relpath%/*}"
+    mkdir -p "$RSRC_ABSDIR/$reldir"
+    cp "$file" "$RSRC_ABSDIR/$reldir/"
+  done
+}
+
+# Add a folder recursively
+# $1 = path prefix (absolute)
+# $2 = relative path to $1 and ${RSRC_ABSDIR} (must not start with /)
+
+function add_foler_recursively {
+  echo "Copying files from folder $1/$2 ..."
+  mkdir -p "${RSRC_ABSDIR}/$2/"
+  cp -R "$1/$2/" "${RSRC_ABSDIR}/$2/"
+}
+
+# Add a single file
+# $1 = path prefix (absolute)
+# $2 = relative path to $1 and ${RSRC_ABSDIR}
+# $3 = file name
+
+function add_single_file {
+  echo "Copying single file $1/$2/$3"
+  mkdir -p "${RSRC_ABSDIR}/$2"
+  cp "$1/$2/$3" "${RSRC_ABSDIR}/$2/"
+}
+
+###### Get filtered list of explicitly installed packages #####
+
+# Note: since both positive and negative filtering makes sense, we do both and require that the result is identical.
+# This ensures people get what they expect.
+
+echo "Create package list"
+
+packages_pos="$(opam list --installed-roots --short --columns=name | grep '^coq\|^menhir\|^gappa')"
+packages_neg="$(opam list --installed-roots --short --columns=name | grep -v '^ocaml\|^opam\|^depext\|^conf\|^lablgtk\|^elpi')"
+
+if [ "$packages_pos" != "$packages_neg" ]
+then
+  echo "The positive and negative list of opam packages differs. Please adjust the package filters!"
+  echo "Positive list = $packages_pos"
+  echo "Negative list = $packages_neg"
+  exit 1
+fi
+
+PRIMARY_PACKAGES="$packages_pos"
+
+###### Associative array with package name -> file filter (regexp pattern) #####
+
+# If not white list regexp is given it is "."
+# If not black list list regexp is given it is "\.byte\.exe$"
+
+declare -A OPAM_FILE_WHITELIST
+declare -A OPAM_FILE_BLACKLIST
+
+OPAM_FILE_WHITELIST[ocaml-variants]='.^' # this has the ocaml compiler in
+OPAM_FILE_WHITELIST[base]='.^' # ocaml stdlib
+OPAM_FILE_WHITELIST[ocaml-compiler-libs]='.^'
+
+OPAM_FILE_WHITELIST[dune]='.^'
+OPAM_FILE_WHITELIST[configurator]='.^'
+OPAM_FILE_WHITELIST[sexplib0]='.^'
+OPAM_FILE_WHITELIST[csexp]='.^'
+OPAM_FILE_WHITELIST[ocamlbuild]='.^'
+OPAM_FILE_WHITELIST[result]='.^'
+OPAM_FILE_WHITELIST[cppo]='.^'
+
+OPAM_FILE_WHITELIST[elpi]='.^' # linked in coq-elpi
+OPAM_FILE_WHITELIST[camlp5]='.^' # linked in elpi
+OPAM_FILE_WHITELIST[ppx_drivers]='.^' # linked in elpi
+OPAM_FILE_WHITELIST[ppxlib]='.^' # linked in elpi
+OPAM_FILE_WHITELIST[ppx_deriving]='.^' # linked in elpi
+OPAM_FILE_WHITELIST[ocaml-migrate-parsetree]='.^' # linked in elpi
+OPAM_FILE_WHITELIST[re]='.^' # linked in elpi
```

Comment:
I don't think we need these to have the `elpi` REPL work, nor the Coq plugin, since both link in there all dependencies.